### PR TITLE
Add schema-support to MysqlAdapter

### PR DESCRIPTION
**Pull request by arjenm**
This is a new PR based on the code of @winkbrace in PR #1292

Its pretty much the same code, modified slightly to work with newer changes. Like winkbrace, I've not touched the PostgreSQL-variant or other variants of this feature.

### CODE REVIEW

File: src/Phinx/Db/Adapter/MysqlAdapter.php [hunk 1]
```diff
@@ -199,19 +199,38 @@ public function hasTable($tableName)
             return true;
         }
 
+        if (strpos($tableName, '.') !== false) {
+            list($schema, $table) = explode('.', $tableName);
+            $exists = $this->queryTableExists($schema, $table);
+            // Only break here on success, because it is possible for table names to contain a dot.
+            if (!empty($exists)) {
```
Comment (dereuromark) [inline src/Phinx/Db/Adapter/MysqlAdapter.php]: Is this a boolean? Then the empty() is a code smell here.

Comment (arjenm) [reply src/Phinx/Db/Adapter/MysqlAdapter.php]: You didn't remark this on his old code :P

So I didn't really bother to change all that, but it seems the queryTableExists actually returns the table name... which is indeed a bit confusing. I'll clean it up a bit.

File: src/Phinx/Db/Adapter/MysqlAdapter.php [hunk 2]
```diff
@@ -201,35 +201,35 @@ public function hasTable($tableName)
...
-        return $this->fetchRow(sprintf(
+        $result = $this->fetchRow(sprintf(
             "SELECT TABLE_NAME
             FROM INFORMATION_SCHEMA.TABLES
             WHERE TABLE_SCHEMA = '%s' AND TABLE_NAME = '%s'",
             $schema,
             $tableName
         ));
+
+        return ! empty($result);
```
Comment (dereuromark) [inline src/Phinx/Db/Adapter/MysqlAdapter.php]: no space after !

Comment (arjenm) [reply src/Phinx/Db/Adapter/MysqlAdapter.php]: argh, working with multiple code styles is confusing :)

Fixed in a forced push

**Comment by arjenm** [comment]
@dereuromark will this pull request be merged?

**Comment by dereuromark** [comment]
From my side ok, @lorenzo ?

**Comment by dereuromark** [comment]
Should this go into master or 0.next?

**Pull request merged**